types/interfaces.ts

```diff
@@ -254,3 +254,9 @@ export enum LogType {
   Request = 'request',
   Misc = 'misc',
 }
+
+export type DirectoryMetadata = {
```

Comment:
Consider changing to `FileOrDirectoryMetadata` or similar, as this is now used for both files and directories.